pynsee/localdata/get_local_data.py

```diff
@@ -108,50 +103,27 @@ def get_local_data(
         
         dataset_version = _find_latest_local_dataset(dataset_version, variables, nivgeo, geocodes[0], update)
        
-    if (not os.path.exists(file_localdata)) or update:
-
-        list_data_all = []       
+    list_data_all = []       
+    
+    for cdg in trange(len(geocodes), desc="Getting data"):
         
-        for cdg in trange(len(geocodes), desc="Getting data"):
-            
-            codegeo = geocodes[cdg]
-            df_default = pd.DataFrame({"CODEGEO": codegeo, "OBS_VALUE": np.nan}, index=[0])
-            
-            try:
-                df = _get_insee_local_onegeo(
-                    variables, dataset_version, nivgeo, codegeo
-                )
-                
-            except Exception as e:
-                #display(e)
-                df = df_default
-
-            list_data_all.append(df)
-
-        data_final = pd.concat(list_data_all).reset_index(drop=True)
+        codegeo = geocodes[cdg]
+        df_default = pd.DataFrame({"CODEGEO": codegeo, "OBS_VALUE": np.nan}, index=[0])
         
-        if data_final.equals(df_default):
-            logger.error("Error or no data found !")            
-        else:
-            data_final.to_pickle(file_localdata)
-            logger.info(f"Data saved: {file_localdata}")
-            
-    else:
         try:
-            data_final = pd.read_pickle(file_localdata)
-        except:
-            os.remove(file_localdata)
-            data_final = get_local_data(
-                variables=variables,
-                dataset_version=dataset_version,
-                nivgeo=nivgeo,
-                geocodes=geocodes,
-                update=True,
-            )
-        else:
-            logger.info(
-                "Locally saved data has been used\n"
-                "Set update=True to trigger an update"
+            df = _get_insee_local_onegeo(
+                variables, dataset_version, nivgeo, codegeo
             )
+            
+        except Exception as e:
+            #display(e)

```

Comment:
remove

Reply:
b27d6751ff1dc1738b522be9dce0680fd43e41d7 should tackle the issue